lwt.opam.template

```diff
@@ -22,3 +22,4 @@ build: [
     "@doc" {with-doc}
   ]
 ]
+available: os-family != "windows" | "ocaml-version" < "5.2"
```

Comment:
Why restrict Windows support? Does this mean Lwt on Windows is only supported on 5.2?

Reply:
To be clear, this is a question for @raphael-proust who added this as part of addressing existing CI failures. (AFAIU, these CI-focused changes are strictly orthogonal to the change targeted by this PR.)

Reply:
Presumably this restriction is causing https://github.com/ocsigen/lwt/actions/runs/11515741400/job/32057007058?pr=1035

Reply:
I added this to the 5.8.0 release, but it's actually not needed because https://github.com/ocsigen/lwt/pull/967 is not merged. (That linked PR fixes some env-related issues on windows, but doing so breaks 5.2 because of some issue with some 5.2 header files.)

We can remove it here and I've actually opened a PR on opam-repo to that end.

Reply:
removed it from here too